### PR TITLE
BI-690 - Edit existing trait which has observation data

### PR DESCRIPTION
**Pull request by nickpalladino**
Possible future improvements (don't want to spend more time on this currently):
- caching editable status
- cancelling editable requests when table row is changed

### CODE REVIEW

**Review by timparsons — APPROVED** [review]

File: src/components/trait/TraitListsTable.vue [hunk 1]
```diff
@@ -268,6 +275,22 @@ export default class TraitTable extends Vue {
...
+  async editable(trait: Trait) {
+    let traitEditable = false;
+    this.loadingTraitEditable = true;
+    try {
+      const [editable] = await TraitService.getTraitEditable(this.activeProgram!.id!, trait.id!) as [boolean, Metadata]
+      traitEditable = editable;
+      this.currentTraitEditable = traitEditable;
+    } catch (error) {
+      // Display error that traits cannot be loaded
+      this.$emit('show-error-notification', 'Error while trying to load traits');
```
Comment (ctucker3) [inline src/components/trait/TraitListsTable.vue:287]: This error message seems incorrect. This is for if the editable endpoint fails right?

**Review by ctucker3 — APPROVED** [review]
One small comment, so approving since I am good with this PR once that is fixed.

File: src/components/trait/TraitListsTable.vue (hunk 1 above)
Comment (nickpalladino) [reply src/components/trait/TraitListsTable.vue:287]: ya, thanks I'll update that

Comment (nickpalladino) [reply src/components/trait/TraitListsTable.vue]: pushed changes

**Pull request merged**